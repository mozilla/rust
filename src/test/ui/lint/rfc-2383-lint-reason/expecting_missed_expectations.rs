// check-pass

#![feature(lint_reasons)]

#[expect(unfulfilled_lint_expectation, reason = "this should catch the nested expect")]
mod foo {
  #[expect(unfulfilled_lint_expectation, reason = "issuing a lint and getting caught above")]
  fn bar() {
    #[expect(unused_mut, reason = "v is unused")]
    let mut v = 0;
  }
}

// make sure it doesn't catch itself
#[expect(unfulfilled_lint_expectation, reason = "this should issue a warning")]
//~^ WARNING this lint expectation is unfulfilled [unfulfilled_lint_expectation]
//~| NOTE `#[warn(unfulfilled_lint_expectation)]` on by default
//~| NOTE this should issue a warning
mod oof {}

fn main() {}
