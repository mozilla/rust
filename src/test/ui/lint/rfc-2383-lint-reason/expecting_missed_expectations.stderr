warning: this lint expectation is unfulfilled
  --> $DIR/expecting_missed_expectations.rs:15:1
   |
LL | #[expect(unfulfilled_lint_expectation, reason = "this should issue a warning")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unfulfilled_lint_expectation)]` on by default
   = note: this should issue a warning

warning: 1 warning emitted

